spec/mongo/auth/cr_spec.rb

```diff
@@ -64,7 +64,9 @@
         }.to raise_error(Mongo::Auth::Unauthorized)
       end
 
-      context 'when compression is used', if: testing_compression? do
+      context 'when compression is used' do
+        require_compression
+        min_server_fcv '3.6'
```

Comment:
Just to make sure I understand, is the idea that going forward we will only check FCV and not server version?

Reply:
There are several failures in the suite if the tests for, say, 3.6 functionality are run on a 3.6 server with 3.4 FCV because the produced errors are different than what a 3.4 server would produce. Requiring FCV instead of server version bypasses these issues. In other words, even though there is support in the test suite to test against clusters where server version != FCV, this is mostly not done currently and thus the min requirement is expressed in FCV and the max requirement in server version.